sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ResourceThrottleRetryPolicy.java

```diff
@@ -144,15 +137,21 @@ private Duration checkIfRetryNeeded(CosmosException dce) {
                     retryDelay = Duration.ofNanos(retryDelay.toNanos() * this.backoffDelayFactor);
                 }
 
+                if (retryDelay.isNegative()) {
+                    // There is no server returned retryAfter
+                    if (Exceptions.isSubStatusCode(dce, HttpConstants.SubStatusCodes.USER_REQUEST_RATE_TOO_LARGE)) {
```

Comment:
Why do we have a distinction in `retryAfter` here for 429/3200 v/s 429/*?

Reply:
Because non-3200 means replication queue full etc. - being too aggressive there could cause harm (you coudl argue if the backend needs to protect against too aggressive retries they should specify a backoff time - but we want to be extra careful here).